motoman_msgs/srv/SelectTool.srv

```diff
@@ -0,0 +1,92 @@
+
+# Changes the tool file used by the controller for (Moto)ROS-controlled motions.
+#
+# NOTE 1: this changes the active tool for (Moto)ROS-controlled motions ONLY.
+#
+#         Regular INFORM programs, jogging or any other (programmed) motions
+#         executed by the controller will NOT use the new tool.
+#
+# NOTE 2: any tool switch initiated by invoking this service WILL however be
+#         taken into account by safety systems such as the FSU and PFL while
+#         executing (Moto)ROS-initiated motions.
+#
+# NOTE 3: active tool will be switched AFTER the controller has completed
+#         execution of the last trajectory segment that was sent to the
+#         controller BEFORE this service was called.
```

Comment:
It's a bit complex, but I believe this captures what we discussed in https://github.com/ros-industrial/motoman/pull/440#issuecomment-992651912 and the follow-up comments.

Reply:
To put another way, "The active tool will not change for any trajectory that has already been submitted to the controller."

Reply:
is that true? Because if IIUC, `motoman_driver` essentially drip-feeds a `JointTrajectoryAction` goal, segment per segment.

MotoROS has an internal buffer, and it accepts new `JointTrajPt`s as long as that buffer is not full. If it is full, it sends a "busy" response to `motoman_driver`, which then keeps trying to queue the current point (ie: the one it couldn't push into the queue).

At any time during this drip-feeding could `SELECT_TOOL` be called.

Reply:
> At any time during this drip-feeding could SELECT_TOOL be called.

That is true. Hence, your message below about the guarantee.